### PR TITLE
Forbid condition sets on bar graphs and add a test

### PR DESCRIPTION
**Pull request by scottbell**
### All Submissions:

* [X] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [X] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [X] Changes address original issue?
* [X] Unit tests included and/or updated with changes?
* [X] Command line build passes?
* [X] Has this been smoke tested?
* [X] Testing instructions included in associated issue?

### CODE REVIEW

**Comment by scottbell** [comment]
This PR addresses: https://github.com/nasa/openmct/issues/4400

File: src/plugins/charts/BarGraphCompositionPolicy.js [hunk 1]
```diff
@@ -23,10 +23,10 @@
 import { BAR_GRAPH_KEY } from './BarGraphConstants';
 
 export default function BarGraphCompositionPolicy(openmct) {
-    function hasRange(metadata) {
+    function hasNonEnumRange(metadata) {
         const rangeValues = metadata.valuesForHints(['range']);
 
-        return rangeValues.length > 0;
+        return rangeValues && rangeValues.length > 0 && rangeValues[0].format !== 'enum';
```
Comment (shefalijoshi) [inline src/plugins/charts/BarGraphCompositionPolicy.js:29]: Perhaps what we should do is check that there is at least 1 range metadata that is Not and enum.
And then in the Vue file, we should filter out any range metadata that is an enum.

**Review by shefalijoshi — CHANGES_REQUESTED** [review]

File: src/plugins/charts/BarGraphCompositionPolicy.js (hunk 1 above)
Comment (scottbell) [reply src/plugins/charts/BarGraphCompositionPolicy.js:29]: If a domain object's range is just:
```
[
    {
        "key": "state",
        "source": "output",
        "name": "State",
        "format": "enum",
        "enumerations": [
            {
                "string": "Foo",
                "value": 0
            },
            {
                "string": "Default",
                "value": 1
            }
        ],
        "hints": {
            "range": 1,
            "priority": 1
        },
        "values": [
            0,
            1
        ],
        "max": null,
        "min": null
    },
    {
        "key": "output",
        "name": "Value",
        "format": "string",
        "hints": {
            "range": 2,
            "priority": 2
        },
        "source": "output"
    }
]
```
Ce should be excluding the first enum, but only keeping the string? Or should we be excluding both?

**Comment by scottbell** [comment]
Talked to @shefalijoshi & @charlesh88 and we're going to just disallow condition sets from being dropped on bar graphs.

**Comment by shefalijoshi** [comment]
Let's revert the changes here to disallow enums and instead, disallow 'conditionSet' type children in the composition policy.

**Review by shefalijoshi — CHANGES_REQUESTED** [review]
Will review once changes are made.

**Review by shefalijoshi — APPROVED** [review]
Looks good, thank you!
### Reviewer Checklist

* [x] Changes appear to address issue?
* [ ] Changes appear not to be breaking changes?
* [ ] Appropriate unit tests included?
* [ ] Code style and in-line documentation are appropriate?
* [ ] Commit messages meet standards?
* [ ] Has associated issue been labelled `unverified`? (only applicable if this PR closes the issue)
* [ ] Has associated issue been labelled `bug`? (only applicable if this PR is for a bug fix)
* [ ] List of Acceptance Tests Performed.

**Pull request merged**